src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -1100,21 +1126,107 @@ subroutine decomp_rates_mimics(bounds, num_soilc, filter_soilc, &
       mimics_cn_r = params_inst%mimics_cn_r
       mimics_cn_k = params_inst%mimics_cn_k
 
+      ! Local column-level ligninNratioAvg because
+      ! not available when use_fates = .true.
+      ! Also get FATES copy of annsum_npp for FATES cases
+      if (use_fates) then
+         ! Initialize
+         do fc = 1,num_soilc
+            c = filter_soilc(fc)
+            ligninNratioAvg_local(c) = 0._r8
+            annsum_npp_col_local(c) = 0._r8
+         end do
+
+         ! Loop over p to get FATES copy of annsum_npp and local ligninNratio,
+         ! both for use_fates = .true.
+         nc = bounds%clump_index
+         do fp = 1, num_soilp
+
+            p = filter_soilp(fp)
+            c = patch%column(p)
+
+            pf = p - col%patchi(c)
+            s  = clm_fates%f2hmap(nc)%hsites(c)
+            annsum_npp(p) = clm_fates%fates(nc)%bc_out(s)%annsum_npp_pa(pf)
+
+            ! Values by pft from Table 1 in Brovkin et al. (2012):
+            ! https://bg.copernicus.org/articles/9/565/2012/bg-9-565-2012.pdf
+            ! TODO Move to the params files? Though this hardwiring will become
+            ! unnecessary when we calc. ligninNratioAvg the same way for FATES
+            ! and non FATES cases.
+            if (patch%itype(p) == noveg) then
```

Comment:
@ekluzek recommends using a switch / case statement instead of if-statement for computational efficiency.

Reply:
also add a simple/short 1x1_brazil test of FATES-MIMICS on izumi to ensure that it keeps working.

Reply:
> @ekluzek recommends using a switch / case statement instead of if-statement for computational efficiency.

I'm still trying to get this to work because the compiler does not consider the integer options that I'm using as constants. I can't change them to parameters in pftconMod because they get calculated at runtime. So my current thought is to convert them to strings for the case statement to work.

Reply:
Oh, these are the integers in pftcon right? There wouldn't be a way to make them into parameters in that case. If you convert to string that makes the evaluations much less efficient. So you should probably go back to the if, else if, structure. Switch statements are good to use if you can, but it's not a requirement. I was worth looking into though.

Reply:
No, instead I will try using `pftname` which is a string already.

Reply:
Yeah, the thing is that the string comparison will be slower than the integer comparison. So although the switch statement might be faster, overall it might be slower, because it has to do a string comparison. I suspect there will be a bigger slowdown in using a string comparison over the gain from using the switch statement. These strings are pretty short so it shouldn't be horrible, but you do potentially have to compare each character individually rather than the value of the integer itself. Another consideration would be readability, but I'm not sure that either one is going to be that much better than the other.

Reply:
...sorry @ekluzek I didn't see that you had responded (github hadn't refreshed).

Since you brought up readability, I do think that the select statement is easier to follow because the strings describe the pfts/cfts explicitly, while in the if-statement they are in shorthand, which isn't 100% intuitive.

Also with the select statement I was able to omit the long use statement listing all the pfts/cfts because now I'm checking explicit strings.

One more thing to consider: This piece of code is temporary. Once FATES and non FATES cases use the same calculation of `ligninNratioAvg`, this code will be deleted.

If you still prefer that we go back to the if-statement, I'm happy to do so. Let me know.

Reply:
> also add a simple/short 1x1_brazil test of FATES-MIMICS on izumi to ensure that it keeps working.

To avoid adding a new test, I'm opting for changing this existing izumi test
`ERS_Ld30.f45_f45_mg37.I2000Clm50FatesRs.clm-FatesColdDefSizeAgeMort`
to running with MIMICS. Let me know if I shouldn't do that.

Reply:
We talked about this and it would be OK to use an existing test that includes MIMICS. The important thing to do there is to make sure MIMICS is included in the testmod name. We do expect that there shouldn't be strange interactions that MIMICS would only work for some FATES options, but not others. In general we expect that in our testing, and you should inspect the code to see if there are unique interactions with other options that would require separate tests for some options. I don't think that's the case for MIMICS though.

Reply:
> make sure MIMICS is included in the testmod name. 

That's taken care of now.

> We do expect that there shouldn't be strange interactions that MIMICS would only work for some FATES options, but not others. In general we expect that in our testing, and you should inspect the code to see if there are unique interactions with other options that would require separate tests for some options. I don't think that's the case for MIMICS though.

With this PR's mods (which allow FATES and MIMICS to work together), I'm pretty sure that you're right @ekluzek